### DISCUSSION TITLE
Which service does it use for routing?

### DISCUSSION THREAD

**Discussion by hellojaccc** [organicmaps/organicmaps#561] [Q&A] [answered]
Which service does it for routing? Graphhopper? OSRM?

**Comment by biodranik** [comment, marked as answer]
We use our own offline routing algorithms.

**Comment by hellojaccc** [comment]
> We use our own offline routing algorithms. 

Thanks for replying. It has the best result among the others! 

I would love to route from web/desktop

**Reply by biodranik** [reply]
What is your use case on the desktop?
You can build Linux/Mac version btw. Maybe even Windows one with some efforts.

**Reply by hellojaccc** [reply]
As i use Linux, I'll look into it!
